shipshape/util/docker/docker.go

```diff
@@ -248,3 +248,36 @@ func Stop(container string, waitTime time.Duration, remove bool) CommandResult {
 
 	return CommandResult{stdout.String(), stderr.String(), err}
 }
+
+// OutOfDate returns true if the image specified
+// has not been pulled recently.
+func OutOfDate(image string) bool {
+	// TODO(ciera): Rather than always return true,
+	// check a file that contains the last time this
+	// image was updated, and only return true if it
+	// is at least N days old.
+	return true
+}
+
+// ImageMatches returns whether the container is running
+// the current version of image.
+func ImageMatches(image, container string) bool {
+	imageHash, err := inspect(image, "{{.Id}}")
+	if err != nil {
+		return false
+	}
+	containerHash, err := inspect(container, "{{.Image}}")
+	if err != nil {
+		return false
+	}
+	return bytes.Equal(imageHash, containerHash)
+}
+
+func inspect(name, format string) ([]byte, error) {
+	var formatter string
+	if len(format) != 0 {
```

Comment:
No, the only uses of inspect so far are these two in ImageMatches. I chose not to have it pass in the property being selected in case we want to select a property with further nesting, or multiple properties. We might be using this later to check for properties like the date an image was built or how long a container has been running.